torch/optim/adam.py

```diff
@@ -402,7 +402,15 @@ def _single_tensor_adam(
                 # Perform stepweight decay
                 param.mul_(1 - lr * weight_decay)
             else:
-                grad = grad.add(param, alpha=weight_decay)
+                # Nested if is necessary to bypass jitscript rules
+                if isinstance(weight_decay, Tensor):
```

Comment:
Since differentiable is already a kwarg, we can use it instead of adding the comment